### PR TITLE
feat(data-driven): set up players and episodes to drive from data

### LINKED ISSUE #16: Consider adding event schema tags for google
https://developers.google.com/search/docs/data-types/event

This will make this look better in google if/when we get indexed

### LINKED ISSUE #63: Update paths, etc
a few things that I think will help with some redirects and also look cleaner...

1. This is really just a practice rather than a tech solution, but the episode files should be named `episode-5.md` vs `Episode-5.md`, as it seems that Gatsby ends up making the latter a redirect to the former anyway, so it's cleaner to remove the need for that redirect.
1. Right now, the path to episodes is `devopspartygames.com/posts/Episode-5/`; it would be nicer if it was like `devopspartygames.com/episodes/episode-5/` etc. We would need to add redirects for the `/posts/xxx.md` existing episodes (easily done in `netlify.toml`, etc) if we changed this, just so older links don't break.

### LINKED ISSUE #68: thought leaderboard footer doesn't have the "community" section
What the leaderboard footer looks like:

image

All the other pages footer:

image

### LINKED ISSUE #79: Make everything more data driven
Mostly just a place for me to capture my thoughts about the refactoring...

- Data element for each player
- Template for an individual player page
- Hosts should be in a data file too

So this means that the episode pages can have stuff in their frontmatter or whatever gatsby calls it that is all data. So like, the actual markdown file for an episode contains almost nothing; right now we manually put in HTML for the list of players with their images/links, the google calendar button, and then later the youtube embed.

We still might do some "manual" stuff (like, the links to jackbox pages for results/images from a game are not consistent enough that I would make it a template thing, but pasting those in as "notes" from the episode is minor compared to the copypasta for everything else).

### LINKED ISSUE #89: New episode template/data needs Google Calendar button
Because the "add to google calendar" comes from a shared calendar, there's not really a programmatic way to make the button appear; I think the best solution is going to be "make a data element that is the google calendar event URI" and shove it in to the template.

The formatting of the google calendar button looks like this:

```
<a target="_blank" href="https://calendar.google.com/event?action=TEMPLATE&amp;tmeid=MzVqZmRkbG1wOWhyaDRuODNqazIzdXA5ZnAgZHUyYXJxZGhlcjJsNGs0MTducXRsdjE4ZmNAZw&amp;tmsrc=du2arqdher2l4k417nqtlv18fc%40group.calendar.google.com"><img border="0" src="/images/add-to-calendar.png" class = "player-episode-page"></a>
<br clear = "all">
```

I think the only thing that changes from event to event is the `href` on the anchor, so if that URI was in data and shoved in here, that should work.

### LINKED ISSUE #90: New episode template needs time/date conversion link
We currently put in a link to the world clock for the time/date of the episode in a couple places, like this:

```
subtitle: Broadcast live at <a href = "https://www.timeanddate.com/worldclock/fixedtime.html?msg=DevOps+Party+Games+S02E02&iso=20210126T20&p1=64" target = "_blank">Tuesday 26 January 8PM US CT</a> 
excerpt: >-
  Streaming live at <a href = "https://www.timeanddate.com/worldclock/fixedtime.html?msg=DevOps+Party+Games+S02E02&iso=20210126T20&p1=64" target = "_blank">20:00 UTC+1</a><br> on Tuesday 26 January
```

I think we can figure this out programmatically from the data and time of the episode from the data; just want to make sure we include it.

### LINKED ISSUE #106: Enhanced "add to calendar" functionality
I slightly suspect that moving away from having to have a Google Calendar for all this nonsense would be helpful.

here's a starting point: https://www.litmus.com/blog/how-to-create-an-add-to-calendar-link-for-your-emails/

we would have to generate .ics files for each episode (there is a web tool to do that) which we could store in "static" and then link to or whatever

### LINKED ISSUE #114: Create page to list all people
Ideally, this should be a list of players (not including hosts) but querying to strip out hosts might be tricky.

We could just call it "Players" and if hosts show up as well, that's not the end of the world

### LINKED ISSUE #117: Fix issue with iCal download
Need to figure out how to make it only download on click of a button, not on load of the episode page

### LINKED ISSUE #122: Create a new default opengraph image for social sharing
Right now we end up pushing the latest episode's social card, which is messy.

we should create a new image that is the default (for non-episode pages)

### LINKED ISSUE #123: Update scoreboard with the new scores structure
The scores now live inside `episodes.json`, so the whole scoreboard component has to be refactored.

### LINKED ISSUE #126: Players with existing scores but also are upcoming players have a total score of 0
There's a bug in the transformer where if a player has a valid score, but also has a list of a score with nothing (i.e., is in an upcoming episode) the total for that player is 0 (look at Jason Yee as an example).

### LINKED ISSUE #127: Add font-awesome
https://fontawesome.com/how-to-use/on-the-web/using-with/react

also, install this plugin

https://www.gatsbyjs.com/plugins/gatsby-plugin-fontawesome-css/

### LINKED ISSUE #128: gatsby-plugin-advanced-sitemap is not compatible with gatsby 3
Plugin gatsby-plugin-advanced-sitemap is not compatible with your gatsby version 3.2.1 - It requires gatsby@^2.0.0

### LINKED ISSUE #129: Player pages have a title of "undefined"
`<title>DevOps Party Games - undefined</title>`

### PR DESCRIPTION
**Pull request by nimbinatus**
Working on updating all of the pages to auto-create based on data files.

Fixes #68 
Fixes #79
Fixes #63
Fixes #89
Fixes #90
Fixes #106
Fixes #114 
Fixes #129
Fixes #128
Fixes #127
Fixes #126
Fixes #16
Fixes #122
Fixes #123
Fixes #117

### CODE REVIEW

**Comment by nimbinatus** [comment]
Sample player page: https://deploy-preview-88--graceful-sage-db0ff.netlify.app/players/laura.santamaria
Sample episode page: https://deploy-preview-88--graceful-sage-db0ff.netlify.app/episodes/s01e02 (note that episodes aren't populated)

**Comment by mattstratton** [comment]
Is this backwards compatible? That is, safe to merge?

**Comment by mattstratton** [comment]
Also, we need to tweak the player image on the players page for mobile image

**Comment by mattstratton** [comment]
Never mind, I'm a dummy and just noticed this was a draft pr. Disregard the "backwards compatible" comment. ;)

**Comment by nimbinatus** [comment]
Yep, just wanted to show a halfway point. Took me way too long to figure out that I had to restart the layouts from scratch for non-Markdown, data-generated pages. Technically, it is safe to merge as I didn't touch the main pages and stood these two categories up separately, but there's still a bunch of fixing needed as you can see...

**Comment by mattstratton** [comment]
No worries - re: the merge thing, there's no rush to merge it (I don't mind having WIP PRs hanging around) I just wanted to clarify what you wanted me to do with the PR, but you clearly marked it as a draft and I just didn't notice :)

**Comment by mattstratton** [comment]
question (keeping in mind this is WIP, I know, just thought of something)

player images will be recreated to just be the player photo, and the name/twitter handle should be in text from the data under their name. If you want to do it that way for the sample, and it will duplicate up, we can do that. Or I can create a `laura-santamaria2.png` for your image that doesn't have the name/handle, and we can use that for dev/testng, etc

File: data/episodes/episodes.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
+{
+  "episodes": [
```
Comment (mattstratton) [inline data/episodes/episodes.json:2]: A couple of other data elements that an episode needs:

- youtube URL (this only should display if it exists; we'll add it after the episode is posted to YT)
- date
- start time
- google calendar URL (should only display if the episode is in the past; could get lazy and make it only display if it exists, and we can delete it from the data when we update with the youtube URL, etc)

**Comment by mattstratton** [comment]
I added a few things. the calendar button thing is just sorta to test it out (might not be even close to ideal) and this has the beginnings of futzing around with the date/time stuff we talked about in chat. 

currently it is displaying the date/time in UTC and kind of doesn't give a rat's patoot about the timezone set.

**Comment by mattstratton** [comment]
The episode that has a calendar link and date/time is this one:

https://deploy-preview-88--graceful-sage-db0ff.netlify.app/episodes/s01e01

**Comment by mattstratton** [comment]
ugh there are conflicts and I don't know what to do about it

**Comment by nimbinatus** [comment]
No worries! I hadn't updated the branch from main, so I bet it's just some slogs by hand. I'll fix them up when I pull it down.

**Comment by mattstratton** [comment]
I made a few updates so that date/time works (pretty much) at this point.

To illustrate, here is an Americas episode: https://deploy-preview-88--graceful-sage-db0ff.netlify.app/episodes/s01e01

And here is an EMEA one: https://deploy-preview-88--graceful-sage-db0ff.netlify.app/episodes/s02e01

I also added some logic to handle episode image (only populated for s01e01 right now) and video stuff too (see s01e01 as well)

google calendar button pretty much works too, but this isn't a great solution (would be better to have an ical get generated I think)

**Comment by mattstratton** [comment]
okay! last push adds a dynamically generated google calendar link (the button on s01e01 still uses the manual one, but the link below is dynamically generated). it seems to work based on local time zone so that's rad.

this addresses #106

**Comment by mattstratton** [comment]
okay so pushed some more updates; the calendar button is now fully dynamically generated (I removed all the old code) so for testing, you can now just click on the button.

**Comment by mattstratton** [comment]
I started thinking a little bit about how "people" work...given that @nimbinatus named it "person" and not "player", I assume the thought is that there will also be a "hosts" field on the episode data entry? 

will the hosts be in the players.json do you think, or will there be a whole separate thought? I don't think there is any need to show "all the episodes a host has hosted" on the person page for sure, although sometimes a host can be a player, so they should show up there. Hmm.

**Comment by mattstratton** [comment]
capturing one of the things we just talked about...

the "episodes listed in `players.json` is deprecated (I assume that we will need to refactor thought leaderboard for this though). instead the way to show a player is on an episode is in the array in the `episodes.json` for that episode.

likewise, `players.json` is really `people.json` and should also have entries for hosts; we can add a field to `episodes.json` for `hosts[]` as well.

**Comment by mattstratton** [comment]
one thing I noticed - it doesn't look like the `episodes[]` field on a player in `players.json` is actually used by anything currently anyway? I don't think the leaderboard uses this. So we probably don't have to deprecate it.

I'm slightly wondering if there's a better way to handle the `scores.json` though; right now you have to create a whole entry for every player. Maybe it could even end up as part of `episodes.json` kind of like this:

```
{
	"episodes": [{
		"id": "1.1.na",
		"date": "2020-06-30T20:00:00-05:00",
		"title": "Season 1 Episode 1",
		"slug": "s01e01",
		"region": "North America",
		"video": "ilWOLddZTBc",
		"image": "episode1.png",
		"players": [{
				"id": "heidi.waterhouse",
				"score": 32
			}, {
				"id": "corey.quinn",
				"score": 12
			},
			{
				"id": "kat.cosgrove",
				"score": 17
			},
			{
				"id": "jeff.smith",
				"score": 22
			},
			{
				"id": "jason.yee",
				"score": 6
			},
			{
				"id": "quintessence.anx",
				"score": 11
			},
			{
				"id": "amy.tobey",
				"score": 9
			},
			{
				"id": "pete.cheslock",
				"score": 1
			}
		],
		"games": [
			"drawful",
			"quiplash"
		]
	}]
}
```

you could just leave score as 0 when creating the episode?

**Comment by mattstratton** [comment]
the sucky part about moving scores into `episodes.json` is the migration of the existing scores (plus of course the refactoring). so it really just depends on where the data entry happens.

I think we could leave it the way it is now, if we wrote some janky bash using `jq` or something to make it easier to populate `scores.json` (it's a slog right now).

**Comment by mattstratton** [comment]
One thing that helps - I still can't make the queries work the way I want, but to filter on players, I tried creating a new `people.json` instead of `players.json` which is structured like this:

```
[
    {
      "id": "corey.quinn",
      "name": "Corey Quinn",
      "twitter": "quinnypig",
      "image": "corey.png",
      "episodes": [
        "s01e01"
      ]
    },
...
```

this lets us filter on `id` which is helpful.

**Comment by mattstratton** [comment]
I'm suspecting that what might have to happen is that we have to define "episode" and "person" and link them in the schema, so we can get a query on the page build for the episode that also includes all the player/host info sorta. it's super messy.

**Comment by mattstratton** [comment]
This is what we need, I think: https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#foreign-key-fields

Except:

```
Keep in mind that in the example above, the link of posts in AuthorJson works because frontmatter and author are both objects. If, for example, the Frontmatter type had a list of authors instead (frontmatter.authors.email), it wouldn’t work since the by argument doesn’t support arrays. In that case, you’d have to provide a custom resolver with Gatsby Type Builders or createResolvers API.
```

**Comment by mattstratton** [comment]
I figured it out! the episode page now properly is loaded players from the component.

still need to finish up all the styling, and all players need new images (that don't have their names/twitter handle)

the new images should be JPG (this will make it easier to not conflict with existing) and should be 500px square.

**Comment by mattstratton** [comment]
to update where we are now...

- I've updated all the episodes and players for all episodes, up to s02e06!
- I did a little refactoring on some of the data; for example, `data/players/players.json` is deprecated and replaced with `data/people/people.json`. Also, I made a change to the structure of `data/episodes/episodes.json` to let gatsby handle it better.
- Players now display on the episode
- Individual players now show their associated episodes
- I made a small change to the leaderboard; it uses the newer people data file, and also it now has a link to the player page for everyone. Since there are people in the file that don't have scores, it only shows people with a score total > 0

things to still work on:
- all the html/css layout is sloppy. I am sorta getting there with the list of players on an episode page, but I'm not thrilled with it
- we need to do something with the games listed; I don't know if it is easy to re-create the verbiage as we use it now; maybe it's just a totally different look. Probably at least want to make the game name link to the Jackbox page for it
- Not all game result links have been populated yet; just a POC right now
- social media/opengraph stuff isn't set up
- update to show a list of all episodes (for homepage and for episode page)
- make a page that shows all people/players with links


*Note: the starting point to check things out is https://deploy-preview-88--graceful-sage-db0ff.netlify.app/episodes/s01e01 - you can click on player images to take you to their player page, etc, and navigate to other episodes either that way, or changing the episode slug, etc*

**Comment by mattstratton** [comment]
okay so this pisses me off.

locally I had to set the date for the Americas episodes to -6:00 now that daylight saving happened.

but on the deploy preview it's off by an hour again

**Comment by mattstratton** [comment]
I am experimenting with adjusting how the `games` key works in the `episodes.json`; the theory is that it's an array for the games, including the link to the game itself as well as this is where the results link would live.

This helps us support non-drawful/quiplash games easier.

It's a little janky, and it would be nice to be able to have something to match the game name to the game link (instead of having to put it in every episode record in the JSON) but it's a start.

**Comment by mattstratton** [comment]
I just added a `gatsby clean` command to the netlify build since it's failing and I think it's a cache issue?

**Comment by mattstratton** [comment]
>I just added a gatsby clean command to the netlify build since it's failing and I think it's a cache issue?

No, the issue was I didn't have a check for something; it works now

**Comment by mattstratton** [comment]
I have refactored the way that the `games` field works on `episodes.json` so that the `GameLink` component works!

**Comment by mattstratton** [comment]
note to review this comment: https://github.com/DevOpsPartyGames/devopspartygames-web/pull/88#issuecomment-797598965

**Comment by mattstratton** [comment]
OK, so I hacked together a new homepage/episode list layout, and also made the /people link work. I'm not super in love with how people are being displayed; I cheated and just stole formatting from episodes, so the image proportions are weird.

I also updated the episode page for how people are shown to use the same styling; I'm not sold that it what they should look like, but it's something to work with :)

**Comment by mattstratton** [comment]
Trying to decide what we should do with the old pages. We can delete them, but then we should create redirects in netlify to point the old episode pages to the correct ones (not a huge deal; just kind of a pain).

We could also just leave them; nothing goes to them, but they'll end up in the sitemap. Although I think there are things we can do with the [plugin](https://www.gatsbyjs.com/plugins/gatsby-plugin-advanced-sitemap/) I'm using to generate the sitemap to exclude those; there shouldn't be anything under the `/posts` URL that we need anymore, so we could remove it from the sitemap.

**Comment by mattstratton** [comment]
I just did a little POC to see how hard it would be (from the episode perspective) to move scores out of their own data file and have them sourced in the `episodes.json` alongside each player.

So it goes from this:

```
[
    {
		"id": "1.1.na",
	    "players": [
	       "heidi.waterhouse",
	       "corey.quinn",
	       "kat.cosgrove",
	       "jeff.smith",
	       "jason.yee",
	       "quintessence.anx",
	       "amy.tobey",
	       "pete.cheslock"
	      ],
		...
	}
]
```
to this ("score" is optional and would be added after the episode is finished):

```
[
    {
		"id": "1.1.na",
	    "players": [
        {
          "id": "heidi.waterhouse",
          "score": 28
        },
        {
          "id": "corey.quinn",
          "score": 28
        },
        {
          "id": "kat.cosgrove",
          "score": 28
        },
        {
          "id": "jeff.smith",
          "score": 28
        },
        {
          "id": "jason.yee",
          "score": 28
        },
        {
          "id": "quitessence.anx",
          "score": 28
        },
        {
          "id": "amy.tobey",
          "score": 28
        },
        {
          "id": "pete.cheslock",
          "score": 28
        }
      ],
		...
	}
]
```

the only change that is needed in the code is the following:

`templates/episode.js`
```
{episode.players.map((player, key) => 
  <PersonDisplay
    personID={player.id}
  />
)}
```

`gatsy-node.js`

```
{
	allEpisodesJson {
		nodes {
			id
			...
			games {
				name
				result
			}
			players {
				id
				score
			}
		}
	}
}
```

I am not committing these changes just yet (since there would need to be updates to all of `episodes.json` to make it work and I don't want to do that data entry until we're sure this is what we want) so that's why I put the notes in here.

I think that in order to make all the things work for the thought leaderboard, we would need to add a new field to `episodes.json` for `season` (we have `region` which can multi-task as "league") but I want to get @nimbinatus's input on if this is feasible to make the scoreboard work in this model? I'm guessing how the scoreboard works would change dramatically with the different data source idea?

**Comment by nimbinatus** [comment]
@mattstratton, yep, I think we could make that work. As long as they're somewhere that's in the GraphQL that we can query, we can find them and do the math for the leaderboard 👍

**Comment by mattstratton** [comment]
@nimbinatus I just pushed up a change that modifies to use the scores/player id in the `players` array on `episodes.json`; all my code is working (I didn't touch the scoreboard).

we still need to add a `season` field to the `episodes.json` but this gives something. It's all up to date as of yesterday's episode with scores (in the `episodes.json` file, that is!)

**Comment by nimbinatus** [comment]
I'm going to work on the scoreboard first to manipulate the data a bit, and I'm going to try to figure out a better way to render the leaderboard so it doesn't rely on live load.

**Comment by mattstratton** [comment]
I added a `season` field to `episodes.json`; I thought it was better to have it as a string, since right now seasons are numbered, but that might not always be the case?

**Comment by nimbinatus** [comment]
@mattstratton are you thinking along the lines of seasons being "Fall 2021"? I had been going with a string like `s01e01` before, but I guess that's not the most readable in the world... (mainly avoiding the potential for misspellings)

**Comment by mattstratton** [comment]
>@mattstratton are you thinking along the lines of seasons being "Fall 2021"? I had been going with a string like s01e01 before, but I guess that's not the most readable in the world... (mainly avoiding the potential for misspellings)

nope, still "1" or "2"; I just thought we wouldn't want to back ourselves into a corner in case we wanted to create a "Special Summer Season" or something. Note: no plans to do any such thing lol

**Comment by mattstratton** [comment]
@nimbinatus feature request in the new board (once you have the fancy graphql and all that)

display rank on the people page? like "current leadboard position: 4/65" or something.

slicing and dicing across seasons/leagues would make figuring this out a little harder but if we think about it from the get-go, maybe not too bad. more about "how do you display this"

**Comment by nimbinatus** [comment]
> display rank on the people page? like "current leadboard position: 4/65" or something.

That was exactly what I was thinking 😄  Someone could link to their player page as a tongue-in-cheek profile of thought leadership.
 
> slicing and dicing across seasons/leagues would make figuring this out a little harder but if we think about it from the get-go, maybe not too bad. more about "how do you display this"

The idea would be if I can get the fancy graphql working, then we can say anything we want, including "Top thought leader in the EMEA region" or something like that.

**Comment by mattstratton** [comment]
Update: we discussed that the idea of "leagues" (as opposed to just being used for the timezone of the episode) are going to be retired after this season, so there is no need to do math on the TLB regarding leagues; it should just be per-season and overall scores.

**Comment by nimbinatus** [comment]
If you look at the GraphiQL with this latest commit, you'll see a bunch of nodes for scores. I'm working on pulling those scores down into the PeopleJson so they're not all separate nodes but appear under the person. That will enable us to do whatever you'd like as it would just be adding another math function and any other fields to the node, and it *should* enable us to get the TLB running without relying on load-time React functions to run it (so no more issues with JS being disabled).

Also, I think the tests are failing because I switched over to Dart-Sass (it was really easy). It builds fine locally :/ If we need to skip that one, we can.

**Comment by nimbinatus** [comment]
Without fail: I commit, and then I get it working :facepalm: There's now two new fields in the PersonJson nodes that hold the total (raw) score and the average score. Any other math can happen in the same fashion in the transformer; just use the `createNodeField` function to call a custom math function you add in the same file at the end to add a different way to calculate the scores.

Next step is to clean up the TLB with a static page for nojs and a filterable page (tabs? dropdown?) for per-season and overall score switches. And this *should* make it easier to add scores to people pages, and we can do the same thing to display rankings in the people pages or an episode page.

The Netlify build is failing because it's on an older NodeJS version (bump up to the latest LTS, 14. It works!).

**Comment by nimbinatus** [comment]
@mattstratton there's a linebreak somewhere: https://answers.netlify.com/t/netlify-node-version-suddenly-not-working/35083

I can look in 10 minutes...

**Comment by mattstratton** [comment]
okay so I fought with `.nvmrc` a bunch and finally got it working (apparently an extra empty line in that file is a problem lol)

**Comment by mattstratton** [comment]
I don't know if this will be a problem?

```
10:40:40 AM: Installing NPM modules using NPM version 6.14.11
10:41:29 AM: npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
10:41:29 AM: > core-js@2.6.12 postinstall /opt/build/repo/node_modules/babel-runtime/node_modules/core-js
10:41:29 AM: > node -e "try{require('./postinstall')}catch(e){}"
10:41:29 AM: > core-js@3.10.0 postinstall /opt/build/repo/node_modules/core-js
10:41:29 AM: > node -e "try{require('./postinstall')}catch(e){}"
10:41:29 AM: > core-js-pure@3.10.0 postinstall /opt/build/repo/node_modules/core-js-pure
10:41:29 AM: > node -e "try{require('./postinstall')}catch(e){}"
10:41:29 AM: > gatsby-telemetry@2.2.0 postinstall /opt/build/repo/node_modules/gatsby-telemetry
10:41:29 AM: > node src/postinstall.js || true
10:41:29 AM: > gatsby-cli@3.2.0 postinstall /opt/build/repo/node_modules/gatsby/node_modules/gatsby-cli
10:41:29 AM: > node scripts/postinstall.js
10:41:30 AM: > gatsby@3.2.1 postinstall /opt/build/repo/node_modules/gatsby
10:41:30 AM: > node scripts/postinstall.js
10:41:32 AM: npm WARN express-graphql@0.9.0 requires a peer of graphql@^14.4.1 but none is installed. You must install peer dependencies yourself.
10:41:32 AM: npm WARN gatsby-plugin-advanced-sitemap@1.6.0 requires a peer of gatsby@^2.0.0 but none is installed. You must install peer dependencies yourself.
10:41:32 AM: npm WARN react-html-parser@2.0.2 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0-0 but none is installed. You must install peer dependencies yourself.
10:41:32 AM: npm WARN react-script-tag@1.1.2 requires a peer of react@^16.0.0 but none is installed. You must install peer dependencies yourself.
```

Maybe `package-lock.json` needs to be re-generated?

**Comment by nimbinatus** [comment]
Hmm... I thought I committed that last night. Let's see how the deploy goes; I've got my laptop up and was juuuust dumping my nvm stuff in. I'll wait to see what happens before messing with it.

**Comment by nimbinatus** [comment]
Ooohhh, wait, no. That's because npm wasn't updated at the same time, apparently. I'm on a new version. It should be fine :)

**Comment by mattstratton** [comment]
I added the capability for a per-season score total; there is now a new `seasons.json` data file that specifies each season, and this adds a new field for the sum for each season for each player.

I started to try to figure out how to modify the scoreboard, but so much of that is based on just pulling in the raw data files, that I was struggling to figure out how to make that use a query instead (we have to use queries now, since the scores aren't in the raw files, etc)

**Comment by mattstratton** [comment]
I broke out the current thought leaderboard into a totally different page, and now it pulls from the new query.

There's a bug in the transformer where if a player has a valid score, but also has a list of a score with nothing (i.e., is in an upcoming episode) the total for that player is 0 (look at Jason Yee as an example).

**Comment by mattstratton** [comment]
I did a bit more; now the individual player pages show their overall ranking (and if they are the top player it shows that). it's not pretty but it is a bit of a POC.

the biggest issue right now is that we can do slice and dice per season, but each season needs its own code right now. I almost feel like making that more dynamic isn't a blocker for release; even if we have to do the "add the extra code each time there is a new season", that should only happen a couple of times a year (not a good overall strategy, but we can release it this way and then start working on making it more maintainable after the release).

I will log an issue for this:

>There's a bug in the transformer where if a player has a valid score, but also has a list of a score with nothing (i.e., is in an upcoming episode) the total for that player is 0 (look at Jason Yee as an example).

because we do need to fix this before release :)

**Comment by nimbinatus** [comment]
I'll fix that bug tonight or tomorrow night. But I will fix it :)

**Comment by mattstratton** [comment]
>I'll fix that bug tonight or tomorrow night. But I will fix it :)

I think I just fixed it with some other improvements :)

**Comment by mattstratton** [comment]
OK I think I'm going to stop for a while; latest push adds the "top season one player" and "top season two player" badges to those people pages; for reference, the top season one player is kat, the top season 2 player is Stormy.

**Comment by mattstratton** [comment]
some reference links:

- [new thought leaderboard](https://deploy-preview-88--graceful-sage-db0ff.netlify.app/thought-leaderboard/)
- [top overall player](https://deploy-preview-88--graceful-sage-db0ff.netlify.app/person/kat.cosgrove) (also top season one player)
- [top season two player](https://deploy-preview-88--graceful-sage-db0ff.netlify.app/person/stormy)

**Comment by mattstratton** [comment]
We need to get the changes from this commit into this branch d41882d0398a03f2ec1a03fb4d336d01d16eaab0

**Comment by mattstratton** [comment]
> We need to get the changes from this commit into this branch [d41882d](https://github.com/DevOpsPartyGames/devopspartygames-web/commit/d41882d0398a03f2ec1a03fb4d336d01d16eaab0)

This was fixed in 1e1643b

Although I'm not sure it's actually working.

**Comment by mattstratton** [comment]
>Although I'm not sure it's actually working.

The favicon is working now.

I did some more player page styling, and also updated the data with the most recent scores, etc.

I also changed the link to the scoreboard in the navigation since I kind of want to have some folks help test things, etc, so that needs to be easier to find :)

**Comment by mattstratton** [comment]
I fixed the issue with the iCal download.

I also updated it so the google calendar/iCal buttons only show up if the episode date is in the future (it is >= current date, so it will work if you build on the date, etc)

**Comment by mattstratton** [comment]
OK so fun fact; I added the compress-images action, but it doesn't seem like it works against forks, which is why this isn't working.

Since we are basically in a place where it's time to go ahead and merge this I'm not going to worry about this for now, but I was thinking in the future, for branch work like this, we should probably have a branch in the repo, rather than against a fork. @nimbinatus if you need us to add you to this org/repo with commit bit, we can do that!

**Pull request merged**